clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -122,15 +122,27 @@ class external_mem_img_desc {
 
   /// size setter
   /// \param [in] img_dims The dimensions of the imported image resource
-  void set_size(sycl::range<3> img_dims) { size = img_dims; }
+  void set_size(sycl::range<3> img_dims) {
+    for (int i = 0; i < 3; i++) {
+      if (img_dims[i] > 1) {
+        size[i] = img_dims[i];
+      }
+    }
+  }
 
   /// image_channel setter
   /// \param [in] img_ch The channel info of the imported image resource
   void set_image_channel(image_channel img_ch) { channel = img_ch; }
 
   /// num_levels setter
   /// \param [in] numLevels The no. of levels in the imported image resource
-  void set_num_levels(unsigned int numLevels) { num_levels = numLevels; }
+  void set_num_levels(unsigned int numLevels) {
```

Comment:
please use a single style for variable names. (please update numLevels).

Reply:
Done